app.rb

```diff
@@ -89,6 +88,49 @@ def list_person_rental
       puts "Date: #{rental_result.date}, Book \"#{rental_result.book.title}\" by #{rental_result.book.author}"
     end
   end
+
+  def save_books
+    File.write('books.json', JSON.pretty_generate(@books.map(&:to_hash)))
+  end
+
+  def save_people
+    File.write('people.json', JSON.pretty_generate(@people.map(&:to_hash)))
+  end
+
+  def save_rentals
+    File.write('rentals.json', JSON.pretty_generate(@rentals.map(&:to_hash)))
+  end
+
+  def load_books
+    return unless File.exist?('books.json')
+
+    JSON.parse(File.read('books.json')).each do |book_data|
+      @books << Book.new(book_data['title'], book_data['author'])
+    end
+  end
+
+  def load_people
+    return unless File.exist?('people.json')
+
+    JSON.parse(File.read('people.json')).each do |person_data|
+      @people << if person_data['classroom']
+                   Student.new(person_data['age'], person_data['name'], person_data['classroom'],
+                               parent_permission: person_data['parent_permission'])
+                 else
+                   Teacher.new(person_data['age'], person_data['name'], person_data['specialization'])
+                 end
+    end
+  end
+
+  def load_rentals
+    return unless File.exist?('rentals.json')
+
+    JSON.parse(File.read('rentals.json')).each do |rental_data|
+      book = @books.find { |b| b.title == rental_data['book']['title'] }
+      person = @people.find { |p| p.id == rental_data['person']['id'] }
+      @rentals << Rental.new(rental_data['date'], book, person) if book && person
```

Comment:
On exit from your app, When I ran the app again, I encountered this error.

![image](https://github.com/CarlosZ96/-Microverse-School-library/assets/78251927/0d646c92-77f3-4c5f-b626-626b0ea775ae)

I checked the saved JSON for rentals and I observed that the issue is according to the name of the retrieved data which should be
`rental_data['person_id'] }` not `rental_data['person']['id'] }`
![image](https://github.com/CarlosZ96/-Microverse-School-library/assets/78251927/670f59d4-4436-49f1-84e5-fab5513c9e09)

Please fix that and ensure that on startup of your app, it loads data from all files🛠🤗